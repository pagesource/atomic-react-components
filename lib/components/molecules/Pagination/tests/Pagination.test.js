import React from 'react';
import { shallow } from 'enzyme';

import { PaginationVanilla } from '../index';

describe('<Pagination />', () => {
  let PaginationComponent = '';
  const props = {
    pageSize: 10,
    data: [],
    onPageChange: jest.fn(),
    className: 'paginate',
    firstLabel: 'First',
    lastLabel: 'Last',
    nextLabel: 'Next',
    previousLabel: 'Previous',
    renderPaginationComponent: jest.fn(),
    totalPosts: 10,
    showNextBtn: true,
    showPreviousBtn: true,
    showLastBtn: true,
    showFirstBtn: true,
    currentPageAria: 'Current Page',
    paginationAriaLabel: 'Pagination',
    nextButtonAria: 'Next Button',
    previousButtonAria: 'Previous Button',
    firstButtonAria: 'First button',
    lastButtonAria: 'Last Button',
    currentPage: 1,
  };
  beforeEach(() => {
    PaginationComponent = shallow(<PaginationVanilla props={props} />);
  });

  test('should render correctly', () => {
    expect(PaginationComponent).toMatchSnapshot();
  });

  it('should render `.pagination` component', () => {
    expect(PaginationComponent.find('.pagination')).toHaveLength(1);
  });

  it('should render an `.last`', () => {
    expect(PaginationComponent.find('.last')).toHaveLength(1);
  });

  it('should render a `firstLabel`', () => {
    expect(PaginationComponent.find('.btn-first').text()).toEqual('First');
  });

  it('should render a `lastLabel`', () => {
    expect(PaginationComponent.find('.last').text()).toEqual('Last');
  });

  it('should test click event on first button', () => {
    PaginationComponent = shallow(<PaginationVanilla {...props}>Sample</PaginationVanilla>);
    PaginationComponent.find('.btn-first').simulate('click');
    expect(props.onPageChange).toHaveBeenCalled();
  });

  it.skip('should test click event on last button', () => {
    PaginationComponent = shallow(<PaginationVanilla {...props} />);
    console.log(PaginationComponent);
    PaginationComponent.find('.last').find('Button').simulate('click');
    expect(props.onPageChange).toHaveBeenCalled();
  });
});
