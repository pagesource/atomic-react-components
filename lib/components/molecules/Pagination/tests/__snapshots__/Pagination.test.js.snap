// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `
<Fragment>
  <nav
    arial-label=""
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="first"
      >
        <Button__StyledButton
          aria-label=""
          className="btn-first"
          disabled={true}
          onClick={[Function]}
        >
          First
        </Button__StyledButton>
      </li>
      <li
        className="previous"
      >
        <Button__StyledButton
          aria-label=""
          disabled={true}
          onClick={[Function]}
        >
          Previous
        </Button__StyledButton>
      </li>
      <PaginationList
        activePage={1}
        currentPageAria=""
        executePagination={[Function]}
        gotoPageAria="Go to Page Number"
        pageNumbers={Array []}
      />
      <li
        className="next"
      >
        <Button__StyledButton
          aria-label=""
          disabled={false}
          onClick={[Function]}
        >
          Next
        </Button__StyledButton>
      </li>
      <li
        className="last"
      >
        <Button__StyledButton
          aria-label=""
          disabled={false}
          onClick={[Function]}
        >
          Last
        </Button__StyledButton>
      </li>
    </ul>
  </nav>
</Fragment>
`;
